src/js/Controllers/RpcFactory.js

```diff
@@ -148,6 +148,65 @@ class RpcFactory {
             }
         })
     }
+    static UIShowResponse(rpc) {
+        var supportedTemplates = capabilities["MEDIA"].displayCapabilities.templatesAvailable;
+        const templateConfiguration = rpc.params.templateConfiguration;
+        const templateExists = templateConfiguration 
+            && templateConfiguration.template ? true : false;
+        // Calculated bool val if the request only tried to change the template layout
+        const onlyChangeTemplate = templateConfiguration !== undefined 
+            && Object.keys(rpc.params).length === 3 //appID, showStrings, templateConfiguration 
+            && rpc.params.showStrings.length === 0
+            && templateExists 
+            && Object.keys(templateConfiguration).length === 1;
+        // Calculated bool value if request only tried to set an unsupported template
+        const unsupportedRequest = templateExists 
+            && onlyChangeTemplate 
+            && !supportedTemplates.includes(templateConfiguration.template);
+        // Calculated bool value if request sets unsupported template value, but other show information was present.
+        const partialSuccess = !unsupportedRequest 
+            && templateExists 
+            && !onlyChangeTemplate 
+            && !supportedTemplates.includes(templateConfiguration.template);
+        
```

Comment:
i think the logic here is a bit confusing, and I think we should simplify it, or at least short circuit the success case.

I think it could boil down to this, but I might be misunderstanding:
```
if (!templateConfiguration.template || supportedTemplates.includes(templateConfiguration.template))
    return "SUCCESS"

if (params.length == 3 && !showStrings.length)
    return "ERROR"

return "WARNINGS"
```